src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -49,4 +50,14 @@
     Optional<Result> findFirstByParticipationIdAndRatedOrderByCompletionDateDesc(Long participationId, boolean rated);
 
     Optional<Result> findDistinctBySubmissionId(Long submissionId);
+
+    /**
+     * This SQL query is used for inserting results if only one unrated result should exist per participation.
+     * This prevents multiple (concurrent) inserts with the same participation_id and rated = 0.
+     *
```

Comment:
Please explain the query in detail for people who do not perfectly understand SQL